commit 11da0236b56bc76a9a1c9b0bbd15cd8737c803a5
Author: Dmitry Rakhchev <rda@emcraft.com>
Date:   Fri Nov 21 23:41:21 2025 +0100

    RM7494: stmmac: correctly setup 1us timer in LPI block
    
    The default value only works for the 100MHz SFR frequency. Calculate
    the correct value and set it during initialization.
    
    Tested on STM32MOP157d.
    
    Without this change when using the direct 1Gbit connection, there are a lot
    of missing outgoing packets.

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c b/drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c
index 188a00065f66..4719f9cf195b 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac4_core.c
@@ -19,13 +19,26 @@
 #include "dwmac4.h"
 #include "dwmac5.h"
 
+static void dwmac4_setup_lpi_1us(struct net_device *dev, void __iomem *ioaddr)
+{
+	struct stmmac_priv *priv = netdev_priv(dev);
+	u32 clk_rate;
+
+	clk_rate = clk_get_rate(priv->plat->stmmac_clk);
+	clk_rate = (clk_rate/1000000) - 1;
+	writel(clk_rate, ioaddr + GMAC_1US_TIC_COUNTER);
+}
+
 static void dwmac4_core_init(struct mac_device_info *hw,
 			     struct net_device *dev)
 {
 	struct stmmac_priv *priv = netdev_priv(dev);
 	void __iomem *ioaddr = hw->pcsr;
-	u32 value = readl(ioaddr + GMAC_CONFIG);
+	u32 value;
 
+	dwmac4_setup_lpi_1us(dev, ioaddr);
+
+	value = readl(ioaddr + GMAC_CONFIG);
 	value |= GMAC_CORE_INIT;
 
 	if (hw->ps) {

